File: examples/python/extract.py
"""Extract structured JSON from a page against a schema.

    pip install ghostcrawl
    export GHOSTCRAWL_API_KEY=YOUR_API_KEY
    python extract.py
"""

import os

from ghostcrawl import GhostcrawlClient

client = GhostcrawlClient(api_key=os.environ["GHOSTCRAWL_API_KEY"])

result = client.page.extract(
    url="https://example.com",
    engine="auto",
    schema={
        "type": "object",
        "properties": {
            "title": {"type": "string"},
            "summary": {"type": "string"},
        },
    },
)
print(result)

File: examples/node/extract.mjs
// Extract structured JSON from a page against a schema.
//
//   npm install @ghostcrawl/sdk
//   export GHOSTCRAWL_API_KEY=YOUR_API_KEY
//   node extract.mjs

import { GhostcrawlClient } from '@ghostcrawl/sdk';

const client = new GhostcrawlClient({ apiKey: process.env.GHOSTCRAWL_API_KEY });

const result = await client.page.extract({
  url: 'https://example.com',
  engine: 'auto',
  schema: {
    type: 'object',
    properties: {
      title: { type: 'string' },
      summary: { type: 'string' },
    },
  },
});
console.log(result);
